client/state/ui/layout-focus/test/reducer.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_SET_FOCUS,
+	LAYOUT_SET_NEXT_FOCUS,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+import layoutFocus from '../reducer';
+
+describe( 'reducer', () => {
+	describe( 'LAYOUT_SET_FOCUS', () => {
+		it( 'sets the current focus area to the passed value', function() {
+			const action = { type: LAYOUT_SET_FOCUS, area: 'sidebar' };
+			const state = layoutFocus( undefined, action );
+			expect( state.current ).to.equal( 'sidebar' );
+		} );
+
+		it( 'sets the previous focus area to the old current value', function() {
+			const action = { type: LAYOUT_SET_FOCUS, area: 'sidebar' };
+			const initialState = { current: 'content', previous: null, next: null };
+			const state = layoutFocus( initialState, action );
+			expect( state.previous ).to.equal( 'content' );
+		} );
+
+		it( 'does not set the current focus area if the value is invalid', function() {
+			const action = { type: LAYOUT_SET_FOCUS, area: 'foobar' };
+			const initialState = { current: 'content', previous: null, next: null };
```

Comment:
I recommend using [`deepFreeze`](https://github.com/substack/deep-freeze) to ensure no accidental mutations of the original object.